File: Java/Algorithm/src/Stack.java

public class Stack {
	static int size=500;
	
	static int[] str=new int[size];
	static int top=0;
	
	public static void main(String[] args) {
		push(1);
		push(2);
		push(3);
		
		System.out.println(pop());
		System.out.println(pop());
		System.out.println(pop());
	}
	
	static int push(int n) {
		if(top>=size)
			return -1;
		str[top++]=n;
		
		return 0;
	}
	
	static int pop() {
		if(top<=0)
			return -1;
		
		return str[--top];
	}
}

File: python/baekjoon/silver4/stack.py
# 백준 10828번 스택
# 20분

n = int(input())
command = []
stack = []

for i in range(n):
    command.append(input())
    
for i in range(n):
    if command[i][:2] == 'pu':
        stack.append(int(command[i][4:]))
    elif command[i][:2] == 'to':
        if stack:
            print(stack[len(stack)-1])
        else:
            print(-1)
    elif command[i][:2] == 'si':
        print(len(stack))
    elif command[i][:2] == 'em':
        if stack:
            print(0)
        else:
            print(1)
    elif command[i][:2] == 'po':
        if stack:
            print(stack.pop())
        else:
            print(-1)